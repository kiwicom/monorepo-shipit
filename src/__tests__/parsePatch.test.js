// @flow

import fs from 'fs';
import path from 'path';

import parsePatch from '../parsePatch';

const FIXTURES_PATH = path.join(__dirname, 'fixtures');
fs.readdirSync(FIXTURES_PATH).forEach(fixtureFile => {
  it(`parses correctly fixture: ${fixtureFile}`, () => {
    const hunks = [];
    for (const hunk of parsePatch(
      fs.readFileSync(path.join(FIXTURES_PATH, fixtureFile), {
        encoding: 'utf8',
      }),
    )) {
      hunks.push(hunk);
    }
    expect(hunks).toMatchSnapshot();
  });
});
