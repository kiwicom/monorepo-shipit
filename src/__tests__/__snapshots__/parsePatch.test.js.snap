// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parses correctly fixture: diff.patch 1`] = `
Array [
  "diff --git a/src/packages/monorepo-shipit/src/Git.js b/src/packages/monorepo-shipit/src/Git.js
index f4d19bd6..05be5559 100644
--- a/src/packages/monorepo-shipit/src/Git.js
+++ b/src/packages/monorepo-shipit/src/Git.js
@@ -9,7 +9,7 @@ function git(args: $ReadOnlyArray<string>) {
 }

 export default class Git {
-  getNativePatchFromID = (revision: string) => {
+  getNativePatchFromID = (revision: string): string => {
     return git([
       'format-patch',
       '--no-renames',
@@ -31,6 +31,10 @@ export default class Git {
     };
   };

+  getChangesetFromNativePatch = (patch: string) => {
+    //
+  };
+
   findDescendantsPath = (
     revision: string,
     roots: $ReadOnlyArray<string>,
",
  "diff --git a/src/packages/monorepo-shipit/src/Sync.js b/src/packages/monorepo-shipit/src/Sync.js
index d45e7901..1892a963 100644
--- a/src/packages/monorepo-shipit/src/Sync.js
+++ b/src/packages/monorepo-shipit/src/Sync.js
@@ -10,14 +10,14 @@ export default class Sync {
   };

   getSourceChangesets = (): Set<Changeset> => {
-    const revision = this.getFirstSourceID();
+    const initialRevision = this.getFirstSourceID();

     const repo = new Git();
     const sourceChangesets = new Set<Changeset>();

-    sourceChangesets.add(repo.getChangesetFromID(revision));
+    sourceChangesets.add(repo.getChangesetFromID(initialRevision));
     repo
-      .findDescendantsPath(revision, ['src/packages/relay'])
+      .findDescendantsPath(initialRevision, ['src/packages/relay'])
       .forEach(revision => {
         sourceChangesets.add(repo.getChangesetFromID(revision));
       });
@@ -37,8 +37,7 @@ export default class Sync {
   run = () => {
     const changesets = this.getFilteredChangesets();

-    console.warn('TODO: push ->', changesets);
-
     // TODO: foreach changes and Git.commitPatch
+    console.warn('Git.commitPatch', changesets);
   };
 }
",
  "diff --git a/src/packages/monorepo-shipit/src/index.js b/src/packages/monorepo-shipit/src/index.js
index 0b901673..41623741 100644
--- a/src/packages/monorepo-shipit/src/index.js
+++ b/src/packages/monorepo-shipit/src/index.js
@@ -2,6 +2,6 @@

 import Sync from './Sync';

-// yarn monorepo-babel-plugin src/packages/monorepo-shipit/src/index.js
+// yarn monorepo-babel-node src/packages/monorepo-shipit/src/index.js

 new Sync().run();
",
]
`;

exports[`parses correctly fixture: diff-eol.patch 1`] = `
Array [
  "diff --git a/test.txt b/test.txt
index a907ec3..3bd1f0e 100644
--- a/test.txt
+++ b/test.txt
@@ -1,2 +1,2 @@
 foo
-bar
\\\\ No newline at end of file
+bar
",
]
`;

exports[`parses correctly fixture: diff-in-diff.patch 1`] = `
Array [
  "diff --git a/src/packages/monorepo-shipit/src/__tests__/fixtures/diff-in-diff.patch b/src/packages/monorepo-shipit/src/__tests__/fixtures/diff-in-diff.patch
index e69de29b..13d590e0 100644
--- a/src/packages/monorepo-shipit/src/__tests__/fixtures/diff-in-diff.patch
+++ b/src/packages/monorepo-shipit/src/__tests__/fixtures/diff-in-diff.patch
@@ -0,0 +1,14 @@
+diff --git a/src/packages/monorepo-shipit/src/__tests__/fixtures/diff.patch b/src/packages/monorepo-shipit/src/__tests__/fixtures/diff.patch
+index 79d62980..8b0893a1 100644
+--- a/src/packages/monorepo-shipit/src/__tests__/fixtures/diff.patch
++++ b/src/packages/monorepo-shipit/src/__tests__/fixtures/diff.patch
+@@ -16,8 +16,7 @@ index f4d19bd6..05be5559 100644
+    };
+
+ +  getChangesetFromNativePatch = (patch: string) => {
+-+    //
+-+  };
+++    // some changes here
+ +
+    findDescendantsPath = (
+      revision: string,
",
  "diff --git a/src/packages/monorepo-shipit/src/__tests__/fixtures/diff.patch b/src/packages/monorepo-shipit/src/__tests__/fixtures/diff.patch
index 79d62980..8b0893a1 100644
--- a/src/packages/monorepo-shipit/src/__tests__/fixtures/diff.patch
+++ b/src/packages/monorepo-shipit/src/__tests__/fixtures/diff.patch
@@ -16,8 +16,7 @@ index f4d19bd6..05be5559 100644
    };

 +  getChangesetFromNativePatch = (patch: string) => {
-+    //
-+  };
++    // some changes here
 +
    findDescendantsPath = (
      revision: string,
",
]
`;
